development/workflow.rst

```diff
@@ -0,0 +1,305 @@
+
+   u/{{username}}/{{topic}}
+
+User branches can be pushed to GitHub to enable collaboration and communication.
+Colleagues should be cognizant that user branches are for prototyping before offering unsolicited code reviews.
+
+Developers can feel free to rebase and force push work to their personal user branches.
+
+A user branch *cannot* be merged into master; it must be converted into a *ticket branch* first.
+
+Ticket branches
+---------------
+
+Ticket branches are associated with a JIRA ticket.
+Only ticket branches can be merged into ``master``.
+
+If the JIRA ticket is named ``DM-NNNN``, then the ticket branch will be named
+
+.. code-block:: text
+
+   tickets/DM-NNNN
+
+A ticket branch can be made by branching off an existing user branch.
+This is a great way to formalize experimental work and shaping it into an LSST software contribution.
+
+When code on a ticket branch is ready for review and merging, follow the :ref:`code review process documentation <git-code-review>`.
+
+.. _git-code-review:
+
+DM Code Review and Merging Process
+==================================
+
+When development on your ticket branch is complete, there is a standard process for reviewing your work and merging it into the ``master`` integration branch.
+
+Review preparation
+------------------
+
+Before submitting your work, we recommend that you organize commits, improve commit messages, and ensure that your work is made against the latest commits on ``master`` with an `interactive rebase <https://help.github.com/articles/about-git-rebase/>`_.
+A common pattern is:
+
+.. code-block:: bash
+
+   git checkout master
+   git pull
+   git checkout tickets/DM-NNNN
+   git rebase -i master
+   # interactive rebase
+   git push --force
+
+Use `Jenkins at ci.lsst.codes <https://ci.lsst.codes/job/stack-os-matrix/build?delay=0sec>`_ to run the Stack's tests with your ticket branch work.
+With Jenkins you can specify the name (or names) of the ticket branches in repositories you are developing; Jenkins will test your work against the ``master`` branch of other Stack repositories by default.
+
+You can monitor builds in the “Bot: Jenkins” HipChat room.
+
+Assign a reviewer
+-----------------
+
+On GitHub, `create a pull request <https://help.github.com/articles/creating-a-pull-request/>`_ for your ticket branch.
+
+.. FIXME policy for naming pull requests?
+
+On your ticket's JIRA page, use the **Workflow** button to switch the ticket's state to **In Review**.
+
+JIRA will ask you to assign reviewers.
+Good reviewers might be experts in the package or domain your are developing.
+But also don't overload a small group of people with code reviews.
+It's also a good idea to distribute your review assignments across the collaboration.
+In particular, asking junior team members to review code is a good way to have them broaden their knowledge of the LSST Stack.
+
+In your message to reviewers, indicate how involved the review work will be ("quick" or a "not quick").
```

Comment:
I think it would be helpful to give a commit hash so that the reviewer can check he's got the version you meant for him to see (e.g., sometimes people forget to push).  To be perfectly clear, I usually dump the output of `git --no-pager log --stat --reverse origin/master..`, though maybe that's overkill.